--- a/drivers/cpufreq/airoha-cpufreq.c
+++ b/drivers/cpufreq/airoha-cpufreq.c
@@ -7,6 +7,9 @@
 #include <linux/pm_domain.h>
 #include <linux/pm_runtime.h>
 #include <linux/slab.h>
+#include <linux/clk-provider.h>
+#include <linux/of_clk.h>
+#include <linux/of.h>
 
 #include "cpufreq-dt.h"
 
@@ -46,6 +49,8 @@ static int airoha_cpufreq_probe(struct p
 	struct airoha_cpufreq_priv *priv;
 	struct device *dev = &pdev->dev;
 	struct device *cpu_dev;
+	struct clk_hw *clk_hw;
+	struct device_node *np;
 	int ret;
 
 	/* CPUs refer to the same OPP table */
@@ -65,9 +70,17 @@ static int airoha_cpufreq_probe(struct p
 	/* Attach PM for OPP */
 	ret = dev_pm_domain_attach_list(cpu_dev, &attach_data,
 					&priv->pd_list);
-	if (ret)
+	if (ret < 0)
 		goto clear_opp_config;
 
+	/* Register this node as a clock provider to satisfy CPU clocks */
+	np = of_find_compatible_node(NULL, NULL, "airoha,en7581-cpufreq");
+	clk_hw = devm_clk_hw_register_fixed_rate(dev, "airoha_dummy_cpu", NULL, 0, 1300000000);
+	if (!IS_ERR(clk_hw) && np) {
+		of_clk_add_hw_provider(np, of_clk_hw_simple_get, clk_hw);
+		of_node_put(np);
+	}
+
 	cpufreq_dt = platform_device_register_simple("cpufreq-dt", -1, NULL, 0);
 	ret = PTR_ERR_OR_ZERO(cpufreq_dt);
 	if (ret) {
